天気課題初回提出致します

ryorta opened the pull request:
天気API課題初回提出致します。

ryorta committed:
天気課題初回提出致します

Yuzunoha commented on js/weather-api/index.js:
```diff
@@ -0,0 +1,44 @@
+    });
+});
+
+  window.addEventListener('DOMContentLoaded', () => {
```
`DOMContentLoaded` のイベントリスナが2個ありますね！可読性を上げるために`DOMContentLoaded` のイベントリスナを一個にまとめましょう。

Yuzunoha commented on js/weather-api/index.js:
```diff
@@ -0,0 +1,44 @@
+
+        const situation = document.getElementById('situation');
+        situation.innerHTML = data['weather'][0]['description'];
+    })
+    });
```
全体的にインデントがずれていますね！インデントのずれはコードレビューで指摘されるので、手動ではなく自動でインデントを揃えましょう。

↓のやり方のショートカット `Command + Shift + B` で、cloud9でjsファイルを自動整形できたのでやってみてください。

Cloud9 で コードの整形
https://blog.nijot.com/etc/post-277/

Yuzunoha commented on js/weather-api/index.js:
```diff
@@ -0,0 +1,44 @@
+        situation.innerHTML = data['weather'][0]['description'];
+    })
+    
+    });
```
インデントを揃えましょう。あと進入禁止マークが出てしまうので、ファイルの末尾で1回改行しましょう。

Yuzunoha commented on js/weather-api/index.js:
```diff
@@ -0,0 +1,44 @@
+window.addEventListener('DOMContentLoaded', () => {
+const placeSelect = document.querySelector('#place');
+    placeSelect.options[0].selected = true;
```
`placeSelect.options[0].selected = true;`
この処理は必要ですか？
必要だよって場合はご返信ください。
必要無ければ消しましょう～。

Yuzunoha commented on js/weather-api/index.js:
```diff
@@ -0,0 +1,44 @@
+});
+
+  window.addEventListener('DOMContentLoaded', () => {
+    fetch('https://api.openweathermap.org/data/2.5/weather?appid=4b5774e9f3d2a07b84f0f2f88e486224&units=metric&lang=ja&q=london')
+    .then(response => response.json())
+    .then(data => {
+
+        const temp = document.getElementById('temp');
+        temp.innerHTML = data['main']['temp'] + '℃';
+
+        const humi = document.getElementById('humi');
+        humi.innerHTML = data['main']['humidity'] + '%';
+
+        const press = document.getElementById('press');
+        press.innerHTML = data['main']['pressure'] + 'hPa';
+
+        const situation = document.getElementById('situation');
+        situation.innerHTML = data['weather'][0]['description'];
+    })
```
この fetch 処理はほぼ同じのが2カ所あるので関数化できますね。共通処理は関数化したほうがよいです。

関数化しないと以下のようなデメリットがあるため。
  - ソースコードが長くなる
  - 修正箇所が増える(修正漏れのリスクが増える)

下記のようにcitiesを引数に取るように関数化しましょう。
```js
const doFetch = (cities) => {
    fetch('https://api.openweathermap.org/data/2.5/weather?appid=4b5774e9f3d2a07b84f0f2f88e486224&units=metric&lang=ja&q=' + cities)
        .then((response) => response.json())
        .then((data) => {
            const temp = document.getElementById('temp');
            temp.innerHTML = data['main']['temp'] + '℃';

            const humi = document.getElementById('humi');
            humi.innerHTML = data['main']['humidity'] + '%';

            const press = document.getElementById('press');
            press.innerHTML = data['main']['pressure'] + 'hPa';

            const situation = document.getElementById('situation');
            situation.innerHTML = data['weather'][0]['description'];
        });
};
```

最初にLondonを表示する処理はこう書きます
```js
window.addEventListener('DOMContentLoaded', () => {
    doFetch('London');
    ....
});
```

Yuzunoha commented on js/weather-api/index.html:
```diff
@@ -7,6 +7,37 @@
+<h1>現在の天気</h1>
+<table>
+    <tr>
+        <th colspan="2">地名</th>
```
地名というヘッダだけあるのは中途半端ですね。

こちらの地名ヘッダに対応するように、選択中のcityをjsで表示するようにしましょう。

ryorta committed:
インデントの整理
fetch処理の関数化
DOMContentLoadedイベントリスナを一つにまとめる
placeSelect.options[0].selected = true;処理の消去
地名ヘッダに対応する都市名を表示

Yuzunoha reviewed (approved):
LGTMです!!!!!
イベントリスナとfetchをまとめることでindex.jsが30行に収まりましたね。
いい感じです！この調子でいきましょう！
